common/models/key-value-model.js

```diff
@@ -1,36 +1,116 @@
+   *
+   * @param {String} key Key to associate with the given value.
+   * @param {*} value Value to persist.
+   * @options {Object} [options]
+   * @property {Number} ttl Time to live for the key-value pair in milliseconds.
+   * @callback {Function} callback The function to execute upon completion.
+   * @param {Error} err Error object.
+   * @promise
+   */
   KeyValueModel.set = function(key, value, options, callback) {
     throwNotAttached(this.modelName, 'set');
   };
 
-  // TODO add api docs
+  /**
+   * Set the TTL (time to live) in ms (milliseconds) for a given key. TTL is the
+   * remaining time before a key-value pair is discarded from the database.
+   *
+   * @param {String} key Key to use when searching the database.
+   * @param {Number} ttl The TTL in ms to set for the key.
+   * @options {Number|Object} [options] Optional settings for the key-value
+   *   pair. If a Number is provided, it is set as the TTL in ms for the
+   *   key-value pair.
+   * @property {Number} ttl TTL in ms for the key-value pair.
+   * @callback {Function} callback The function to execute upon completion.
+   * @param {Error} err Error object.
+   * @param {Number} ttl TTL in ms.
+   * @promise
+   */
   KeyValueModel.expire = function(key, ttl, options, callback) {
     throwNotAttached(this.modelName, 'expire');
   };
 
-  // TODO add api docs
+  /**
+   * Return the TTL (time to live) for a given key. TTL is the remaining time
+   * before a key-value pair is discarded from the database.
+   *
+   * @param {String} key The key to use when searching the database.
+   * @options {Object} options Placeholder for optional settings.
+   * @callback {Function} callback The function to execute upon completion.
+   * @promise
+   */
   KeyValueModel.ttl = function(key, options, callback) {
     throwNotAttached(this.modelName, 'ttl');
   };
 
-  // TODO add api docs
+  /**
+   * Return all keys for the given filter.
+   *
+   * **WARNING**: This method is not suitable for large data sets as all
+   * key-values pairs are loaded into memory at once. For large data sets,
+   * use `iterateKeys()` instead.
+   *
+   * @param {Array.<String>} filter The key(s) to use when searching the
+   *   database.
+   * @options {Object} [options] Placeholder for optional settings.
+   * @callback {Function} callback The function to execute upon completion.
+   * @promise
+   */
   KeyValueModel.keys = function(filter, options, callback) {
     throwNotAttached(this.modelName, 'keys');
   };
 
-  // TODO add api docs
+  /**
+   * Return all keys for the given filter. Similar to `.keys()` but instead
+   * allows for iteration over a large data set without having to load
+   * everything into memory at once.
+   *
+   * @param {Array.<String>} filter The key to use when searching the database.
+   * @options {Object} [options] Placeholder for optional settings.
+   * @callback {Function} callback The function to execute upon completion.
```

Comment:
There is not callback and no promise is returned. Instead, the method returns an instance of `AsyncIterator` as described in juggler's API.